### ISSUE TITLE
Training requirements

### ISSUE THREAD

**Issue by TrickyJustice** [kwsong0113/diffusion-forcing-transformer#3]
Hello,
Thank you for the amazing work! I had a doubt regarding training dfot. Do you think I can train this model on 40 GB VRAM A100s? I tried training the dfot-video architecture on a small dataset but it always encounters OOM error even with very small batch size.

**Comment by buoyancy99** [comment]
As reported in our original paper, most of our results are using 12xH100 GPUs. However, this shall not be absolutely needed, as our model converges way before our final training step and batch size can be smaller. Could you let use know how many A100s do you have? Here are a few things you can do when you have a small compute budget:

1. use fewer GPUs & smaller batch size, reduce learning rate proportionally (or square root of rate) and train longer
2. remove the ema (we used it because everyone uses it to increase diffusion performance, but this is at a cost of convergence & memory)
3. Depending on how many GPUs you have (could you let me know), I can quickly give you a branch that uses DeepSpeed to shard optimizer states, which can significantly reduce GPU memory at minor training speed cost

**Comment by TrickyJustice** [comment]
I am currently working with 8 40GB A100s. But I have made some changes in the code though. As external conditioning, I use CLIP model to generate text conditionings. I have made the required changes in the DiT3D structure and in the training process in dfot_video.py
Here's the problem though: If I increase the subdataset size to more than 4, I encounter OOM error even if I use DiT-S without any external conditionings. So I start training with subdataset size=4 and keep the batch sizes small enough (4) to start the training. But as soon as the training starts, the "Validation DataLoader" tqdm starts? Is this normal? 

<img width="1305" alt="Image" src="https://github.com/user-attachments/assets/95be4c76-6824-45cb-ab43-13232bc597c0" />

Thanks a lot for your fast response though!

**Comment by buoyancy99** [comment]
Validation at the beginning is expected, it's check to make sure your validation code doesn't error out.

If you have memory issues, you should be able to use deepspeed stage 1 or stage 2 with ~2 lines of code change following. This will distribute the optimizer states & gradient to different GPUs to save memory.

 https://lightning.ai/docs/pytorch/stable/advanced/model_parallel/deepspeed.html#deepspeed-zero-stage-2

**Comment by TrickyJustice** [comment]
Thanks for this. I will try this out. I also noticed that training_schedule is not in the configuration structure anywhere but you call it in continuous diffusion

**Comment by TrickyJustice** [comment]
I also don't see precond_scale in the configs. These variables are used in continuous diffusion.

**Comment by kwsong0113** [comment]
Those configurations are located in the `configurations/data_experiment` folder (e.g., `realestate10k_video_generation.yaml`). You can learn how the configuration hierarchy works in our [wiki](https://github.com/kwsong0113/diffusion-forcing-transformer/wiki/Configuration-Hierarchy). I'd be happy to provide further assistance if you share the specific commands you're using!

**Comment by TrickyJustice** [comment]
Yeah I found them. But when I train the model, lets say for the code you provide: python -m main +name=MCRAFT dataset=minecraft algorithm=dfot_video experiment=video_generation @diffusion/continuous @DiT/B
It throws an error stating training_scheduler is not in the structure. I figure that might be because the dfot_video.yaml under algorithms does not have these variables defined there.

**Comment by kwsong0113** [comment]
The command works on my end in the latest commit. Have you made any changes to the code?
